typing/typecore.ml

```diff
@@ -2517,56 +2517,58 @@ let generalize_and_check_univars env kind exp ty_expected vars =
   List.iter generalize vars;
   check_univars env kind exp ty_expected vars
 
-(* [check_partial_application] implements both the [non-unit-statemnt] and the
-   [ignored-partial-application] warnings.
+(* [check_statement] implements the [non-unit-statement] check.
 
-   [non-unit-statemnt] verifies that [exp] has type [unit]. It is applied in
-   contexts where their value of [exp] is known to be discarded (eg the lhs of a
-   sequence).
+   It is checked that [exp] has type [unit]. This check is applied in contexts
```

Comment:
"It is checked that [exp] has type [unit]" is not fully clear to me. "It checks that [exp] has type [unit]" (or "we check") is clearer.

Proposal:

> This check is called in contexts where the value of the expression is known to be discarded
> (eg. the lhs of a sequence). We check that [exp] has type unit, or has an explicit type annotation;
> otherwise we raise the [non-unit-statement] warning.

Reply:
Amended as suggested, thanks!